estate/models/estate_property.py

```diff
@@ -0,0 +1,86 @@
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+from datetime import date, timedelta
+
+
+class EstateProperty(models.Model):
+    _name = "estate_property"
+    _description = "An Estate Property"
+    _order = "id DESC"
+
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, default=date.today() + timedelta(days=90))
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer(string="Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        [
+            ('north', 'North'),
+            ('west', 'West'),
+            ('east', 'East'),
+            ('south', 'South')
+        ]
+    )
+    active = fields.Boolean("Active", default=True)
+    state = fields.Selection([
+        ('new', 'New'),
+        ('received', 'Offer Received'),
+        ('accepted', 'Offer Accepted'),
+        ('sold', 'Sold'),
+        ('cancelled', 'Cancelled')],
+        default='new',
+        required=True
+    )
+    property_type_id = fields.Many2one('property_type', string='Type')
+    user_id = fields.Many2one('res.users', string='Salesperson', default=lambda self: self.env.user)
+    buyer_id = fields.Many2one('res.partner', copy=False)
+    tags_ids = fields.Many2many("property_tag", string="Tags")
+
+    offer_ids = fields.One2many("property_offer", "property_id", string="Offers")
+    total_area = fields.Float(compute="_compute_area")
+    best_offer = fields.Float(compute="_compute_best_offer")
+
+    @api.depends("garden_area", "living_area")
+    def _compute_area(self):
+        for record in self:
```

Comment:
General comments for loops like this: 
```suggestion
        for estate in self:
```
`record` is a bit too generic, it's fine for small method like this. But when you'll work in huuuge methods, you'll be happy to have a name a bit more meaningful trust me